checks/datadog.py

```diff
@@ -72,6 +50,49 @@ def __init__(self, logger, dogstreams):
         self.logger = logger
         self.dogstreams = dogstreams
 
+    @classmethod
+    def _instantiate_dogstreams(cls, logger, config, dogstreams_config):
+        """
+        Expecting dogstreams config value to look like:
+           <dogstream value>, <dog stream value>, ...
+        Where <dogstream value> looks like:
+           <log path>
+        or
+           <log path>:<module>:<parser function>
+        """
+        dogstreams = []
+        # Create a Dogstream object for each <dogstream value>
+        for config_item in dogstreams_config.split(','):
+            try:
+                config_item = config_item.strip()
+                parts = windows_friendly_colon_split(config_item)
+                if len(parts) == 1:
+                    # If the dogstream includes a wildcard, we'll add every
+                    # matching path.
+                    for path in cls._get_dogstream_log_paths(parts[0]):
+                        dogstreams.append(Dogstream.init(logger, log_path=path))
+                elif len(parts) == 2:
+                    logger.warn("Invalid dogstream: %s" % ':'.join(parts))
+                elif len(parts) >= 3:
+                    dogstreams.append(Dogstream.init(
+                        logger,
+                        log_path=parts[0],
```

Comment:
That would be awesome, Yann. Thanks!
On Jun 5, 2015 3:01 PM, "Yann" <notifications@github.com> wrote:

> In checks/datadog.py
> <https://github.com/DataDog/dd-agent/pull/1550#discussion_r31855564>:
>
> > +        # Create a Dogstream object for each <dogstream value>
> > +        for config_item in dogstreams_config.split(','):
> > +            try:
> > +                config_item = config_item.strip()
> > +                parts = windows_friendly_colon_split(config_item)
> > +                if len(parts) == 1:
> > +                    # If the dogstream includes a wildcard, we'll add every
> > +                    # matching path.
> > +                    for path in cls._get_dogstream_log_paths(parts[0]):
> > +                        dogstreams.append(Dogstream.init(logger, log_path=path))
> > +                elif len(parts) == 2:
> > +                    logger.warn("Invalid dogstream: %s" % ':'.join(parts))
> > +                elif len(parts) >= 3:
> > +                    dogstreams.append(Dogstream.init(
> > +                        logger,
> > +                        log_path=parts[0],
>
> Thanks for the quick reply :) If you don't mind I'll rebase your PR on top
> of master and bring your logic to this part of the code too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DataDog/dd-agent/pull/1550/files#r31855564>.
>